modules/xfem/include/efa/EFAFragment2D.h

```diff
@@ -51,5 +55,59 @@ class EFAFragment2D : public EFAFragment
   void addEdge(EFAEdge * new_edge);
   std::set<EFANode *> getEdgeNodes(unsigned int edge_id) const;
   EFAElement2D * getHostElement() const;
+  /**
+   * Splits fragment into many fragments depending on cuts
+   * @param EmbeddedNodes is used to create new intersection nodes in the case of cut plane
+   * intersections
+   * @return vector of fragments
+   */
+  std::vector<EFAFragment2D *> split(std::map<unsigned int, EFANode *> & EmbeddedNodes);
+  /**
+   * Splits fragment into many fragments depending on cuts
+   * @param EmbeddedNodes is used to create new intersection nodes in the case of cut plane
+   * intersections
+   * @param vector of fragment cut plane ids
+   * @param vector of vectors with each cut node binned by cut planes
+   * @return vector of fragments
+   */
+  std::vector<EFAFragment2D *> split(std::map<unsigned int, EFANode *> & EmbeddedNodes,
+                                     std::vector<unsigned int> frag_cut_plane_idx,
+                                     std::vector<std::vector<EFANode *>> frag_cut_plane_nodes);
+  /**
+   * Determine if two lines intersect
+   * @param two length vector with coordinates of point 1
+   * @param two length vector with coordinates of point 2
+   * @param two length vector with coordinates of point 1 of cutting line
+   * @param two length vector with coordinates of point 2 of cutting line
+   * @param two length vector with coordinates of intersection point if it exists
+   * @return bool indicating if lines intersect
+   */
+  bool IntersectSegmentWithCutLine(const std::vector<double> & segment_point1,
```

Comment:
A few of these methods already exist in XFEMFuncs.C. Unless there's a really good reason to reimplement them here, you should use those, or add missing functionality to them if needed.